core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -634,6 +636,10 @@ boolean isPreserveAliases() {
                 if (!noIndicesAllowedOrMatches(options, matches)) {
                     throw infe(expression);
                 }
+
+                if (expression.contains("*")) {
```

Comment:
> That would resolve to: ["test1", "test2", "test3", "-test2"], assuming a -test2 index exists, which I think is the correct behavior?

I see, yea that's ok, just different compared to before (we would previously add and remove `test2`, but there was no way to refer to `-test2` if existing I believe), but correct. Do we have a test for this too?

I think documenting all this under the api conventions page would clear things up.